salt/client/__init__.py

```diff
@@ -1493,10 +1493,6 @@ def pub(self,
                                                  master_uri=master_uri)
 
         try:
-            # Ensure that the event subscriber is connected.
-            # If not, we won't get a response, so error out
-            if not self.event.connect_pub(timeout=timeout):
-                raise SaltReqTimeoutError()
```

Comment:
Good analysis. I suspect that the test suite will be tolerant of this change but this is something to keep in mind. I'd like to get this in and see how things go.

Thanks @DmitryKuzmenko and @skizunov as always.